### ISSUE TITLE
Semantic conventions should use int or double, not number

### ISSUE THREAD

**Issue by anuraaga** [open-telemetry/opentelemetry-specification#1274]
**What are you trying to achieve?**

Easy mapping from semantic conventions to OTLP / sane handling in OpenTelemetry collector.

**Additional context.**

In the collector, we have logic to parse HTTP status code

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/master/exporter/awsxrayexporter/translator/http.go#L58

It calls `IntVal()` which is the `pdata`, similar to OTLP, representation of data in the collector. But since the semantic convention only specifies number

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/http.md#common-attributes

It seems like currently, the only way to properly parse a number is to check both intval and doubleval. Is this intended? I guess the semantic convention types are supposed to match the OTLP types. This came up since currently JS populates all numeric attributes as double - this is probably because JS uses double for all numbers. This seems wrong, but with the spec as is, it technically is ok.

**Comment by Oberon00** [comment]
The related build-tools issue: https://github.com/open-telemetry/build-tools/issues/13

**Comment by Oberon00** [comment]
> It seems like currently, the only way to properly parse a number is to check both intval and doubleval. Is this intended? I guess the semantic convention types are supposed to match the OTLP types.

No, the semantic convention types are not bound by OTLP. Rather they should correspond to what is allowed in the specification for <https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/common/common.md#attributes>. Since OTLP is modeled after the same source, this is a non-issue today, but when the generator tool was written, #425 was not resolved and we only that the type "numeric" in the spec, not int64/double.

This is the historical explanation for how we ended up only having "number", I agree this should be improved to distinguish int64/double.

**Comment by ocelotl** [comment]
I asked Claude to review this issue and Claude suggests it be closed with this explanation:

> The requested distinction now exists: specification/common/README.md defines attribute primitives as 'double precision floating point (IEEE 754-1985), or signed 64 bit integer' with separate Integers and Floating Point Numbers sections, and semantic conventions (moved to open-telemetry/semantic-conventions) now use explicit int/double types. This resolves the historical 'number'-only limitation Oberon00 described (from unresolved #425 / build-tools#13).

To guard against a mistaken suggestion, Claude then independently re-verified this conclusion:

> Core ask done: semantic-conventions model now uses type:int (e.g. the reporter's own http.response.status_code) not "number"; build-tools#13 (int/double generator) CLOSED 2021-03-16 and root #425 CLOSED; common/README.md defines distinct int64/double primitives. Stale OPEN issue, no closing PR linked but fully superseded.

Claude evaluated this issue with high confidence that it can be closed.

Claude has classified this issue as resolved.

The intention of this comment is to make it easier for a human being to review this issue before making a decision.
